### ISSUE TITLE
Option to set charge speed for Min SoC + PV Mode

### ISSUE THREAD

**Issue by idominiki** [pkuehnel/TeslaSolarCharger#1126]
Would be great if a charge speed can be set while in Min SoC + PV Mode

**Comment by pkuehnel** [comment]
Can you give me an example when this would make sense?

**Comment by idominiki** [comment]
Network-friendly charging 😊

**Comment by pkuehnel** [comment]
You can change the max charging speed in the car configuration page but does not make sense imo, as charging is more efficient the faster you charge (at least with Tesla's as they have 300W standby consumption).

**Comment by pdb01** [comment]
I have the exact same wish as I live in flanders (Belgium), where the grid operator has imposed a fee based on the quarterly peak power you ask from grid (in a way to limit peak usage and reduce investment needs). Concretely: we have to pay 40 EUR for each kW above 2,5 kW on the yearly invoice, which is measured on quarterly basis and the "worst" quarter of the year (averaged) is taken into account for this fee.

**Therefore, I'm looking indeed as well to have a way that TSC looks at the max combined current (and in case of Spot Price + PV also calculates the charging slots based on the reduced max combined current...)?**

Even better for my use case would be that TSC looks to the current grid power, which I put for the moment at e.g. -2500W (as the min. peak value) in order that TSC always charges at max. 2500W, compensates in case another device (oven, heater, ...) powers up and allows higher car charging when solar power is active.
This works great for Min SoC + PV (with SoC at 0%), but does not work for automated slot planning with Spot price.... 

thanks anyway, the tool is great already, and especially with Swagger integration I'm able to automate parts of it :-)

**Comment by pkuehnel** [comment]
So @idominiki what @pdb01 wants is basically the same as what you want? Reducing the load on the grid?
@pdb01In your case it is based as a sum of kW on all phases. How is this measured? If you are 2 seconds above this threshold you have to pay? And FYI: Do not use the UpdateBaseConfiguration endpoint frequently as calling this results in many things going on under the hood. That's why there is a separate `UpdatePowerBuffer` endpoint. This just updates the value in memory and you can use it as often as you want.

**Comment by idominiki** [comment]
@pkuehnel  of course this is the main reason. If I want to charge at full power to battery level XX, I use the full power mode.

**Comment by pdb01** [comment]
No, it's an averaged measurement. The average power demand of the current 15-minutes interval is remembered for each month. In the end, an average of the 12 months peak power values is then the one used for invoicing.
Anyways, I'm only at 1 phase.

I intend to use UpdateCarConfiguration to limit the car current to a "safe" value when I use a home assistant switch for night time charging with low spot prices. But it is a challenge... a maximum charge speed would make things easier :-)
